apps/right-to-rent-check/behaviours/tenants.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const _ = require('lodash');
+
+module.exports = fields => {
+
+  return superclass => class extends superclass {
+
+    saveValues(req, res, callback) {
+      const name = req.body['tenant-name'];
+      const tenants = req.sessionModel.get('tenants') || [];
+
+      let tenant = _.find(tenants, {'tenant-name': name});
+      if (!tenant) {
+        tenant = fields.length ? _.pick(req.body, fields) : req.body;
+        if (!_.isEmpty(tenant)) {
```

Comment:
I see. You'd confused me by redefining `tenant` from where it was defined as the result of the `find` operation on the array.

I'd maybe use a different name for the new object constructed from `req.body` to prevent that confusion.